Crash: java.lang.NullPointerException com.cooper.wheellog.LoggingService.onDestroy

Occurrences: 89, impacted users: 48
Affected: Android 9, 10, 11
First occurrence: apk 70 (5 Sep), apk 71 still affected

java.lang.RuntimeException: 
  at android.app.ActivityThread.handleStopService (ActivityThread.java:4583)
  at android.app.ActivityThread.access$2100 (ActivityThread.java:274)
  at android.app.ActivityThread$H.handleMessage (ActivityThread.java:2139)
  at android.os.Handler.dispatchMessage (Handler.java:107)
  at android.os.Looper.loop (Looper.java:237)
  at android.app.ActivityThread.main (ActivityThread.java:8167)
  at java.lang.reflect.Method.invoke (Native Method)
  at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run (RuntimeInit.java:496)
  at com.android.internal.os.ZygoteInit.main (ZygoteInit.java:1100)
Caused by: java.lang.NullPointerException: 
  at com.cooper.wheellog.LoggingService.onDestroy (LoggingService.java:188)
  at android.app.ActivityThread.handleStopService (ActivityThread.java:4563)

--- a/app/src/main/java/com/cooper/wheellog/LoggingService.java
+++ b/app/src/main/java/com/cooper/wheellog/LoggingService.java
@@ -183,8 +183,12 @@ public class LoggingService extends Service
     @SuppressWarnings("MissingPermission")
     @Override
     public void onDestroy() {
-        String path = fileUtil.getAbsolutePath();
-        fileUtil.close();
+        String path = "";
+
+        if (fileUtil != null) {
+            path = fileUtil.getAbsolutePath();
+            fileUtil.close();
+        }
 
         if (!isNullOrEmpty(path)) {
             Intent serviceIntent = new Intent(Constants.ACTION_LOGGING_SERVICE_TOGGLED);

